File: work/fix-encoding.mjs
import fs from "fs";

// Fix FeedbackPanel.tsx
let content = fs.readFileSync("C:/Users/wei.gu/Documents/demogo/web/src/components/dashboard/FeedbackPanel.tsx", "utf8");
content = content.replace('          ????\n          <select className="select" value={type}', '          问题类型\n          <select className="select" value={type}');
content = content.replace('          ????\n          <select className="select" value={demoId}', '          关联项目\n          <select className="select" value={demoId}');
content = content.replace('          ????\n          <textarea className="textarea"', '          问题描述\n          <textarea className="textarea"');
fs.writeFileSync("C:/Users/wei.gu/Documents/demogo/web/src/components/dashboard/FeedbackPanel.tsx", content, "utf8");
console.log("FeedbackPanel.tsx done");

// Fix FormHostingPanel.tsx
content = fs.readFileSync("C:/Users/wei.gu/Documents/demogo/web/src/pages/dashboard/FormHostingPanel.tsx", "utf8");
content = content.replace("<p>DemoGo ??????????????????????????</p>", "<p>DemoGo 自动识别项目中的表单字段，帮你收集用户信息。</p>");
content = content.replace("<p>??? {form.submissionCount || 0} ??? ? ??? {formatDate(form.createdAt)}</p>", "<p>已收到 {form.submissionCount || 0} 条提交 · {formatDate(form.createdAt)}</p>");
content = content.replace('<Badge tone="success">??</Badge>', '<Badge tone="success">收集中</Badge>');
fs.writeFileSync("C:/Users/wei.gu/Documents/demogo/web/src/pages/dashboard/FormHostingPanel.tsx", content, "utf8");
console.log("FormHostingPanel.tsx done");

// Fix ProjectDetail.tsx
content = fs.readFileSync("C:/Users/wei.gu/Documents/demogo/web/src/pages/dashboard/ProjectDetail.tsx", "utf8");
content = content.replace("<dt>????</dt><dd>{demo.deploySourceLabel", "<dt>发布来源</dt><dd>{demo.deploySourceLabel");
content = content.replace("<dt>????</dt><dd>{demo.hostingModeLabel", "<dt>托管方式</dt><dd>{demo.hostingModeLabel");
content = content.replace("<dt>????</dt><dd>{demo.runtime?.statusLabel", "<dt>运行状态</dt><dd>{demo.runtime?.statusLabel");
fs.writeFileSync("C:/Users/wei.gu/Documents/demogo/web/src/pages/dashboard/ProjectDetail.tsx", content, "utf8");
console.log("ProjectDetail.tsx done");

console.log("All 3 files fixed.");

File: work/fix_encoding.py
import os

base = r"C:\Users\wei.gu\Documents\demogo\web\src"

def fix_file(rel_path, replacements):
    full_path = os.path.join(base, rel_path)
    with open(full_path, 'r', encoding='utf-8') as f:
        content = f.read()
    for old, new in replacements:
        if old not in content:
            print(f"  WARNING: pattern not found in {rel_path}: {repr(old[:40])}")
        content = content.replace(old, new)
    with open(full_path, 'w', encoding='utf-8') as f:
        f.write(content)
    print(f"  Fixed: {rel_path}")

# FeedbackPanel.tsx
fix_file(r"components\dashboard\FeedbackPanel.tsx", [
    ("          ????\r\n          <select className=\"select\" value={type}",
     "          问题类型\r\n          <select className=\"select\" value={type}"),
    ("          ????\r\n          <select className=\"select\" value={demoId}",
     "          关联项目\r\n          <select className=\"select\" value={demoId}"),
    ("          ????\r\n          <textarea className=\"textarea\"",
     "          问题描述\r\n          <textarea className=\"textarea\""),
])

# FormHostingPanel.tsx
fix_file(r"pages\dashboard\FormHostingPanel.tsx", [
    ("<p>DemoGo ??????????????????????????</p>",
     "<p>DemoGo 自动识别项目中的表单字段，帮你收集用户信息。</p>"),
    ("<p>??? {form.submissionCount || 0} ??? ? ??? {formatDate(form.createdAt)}</p>",
     "<p>已收到 {form.submissionCount || 0} 条提交 · {formatDate(form.createdAt)}</p>"),
    ("<Badge tone=\"success\">??</Badge>",
     "<Badge tone=\"success\">收集中</Badge>"),
])

# ProjectDetail.tsx
fix_file(r"pages\dashboard\ProjectDetail.tsx", [
    ("<dt>????</dt><dd>{demo.deploySourceLabel",
     "<dt>发布来源</dt><dd>{demo.deploySourceLabel"),
    ("<dt>????</dt><dd>{demo.hostingModeLabel",
     "<dt>托管方式</dt><dd>{demo.hostingModeLabel"),
    ("<dt>????</dt><dd>{demo.runtime?.statusLabel",
     "<dt>运行状态</dt><dd>{demo.runtime?.statusLabel"),
])

print("All done!")
